sslh-select.c

```diff
@@ -319,6 +408,23 @@ void main_loop(int listen_sockets[], int num_addr_listen)
                             check_access_rights(in_socket, cnx[i].proto->service)) {
                             tidy_connection(&cnx[i], &fds_r, &fds_w);
                             res = -1;
+                        } else if (cnx[i].proto->fork) {
+                            if (!fork()) {
+                                struct connection *pcnx_i = &cnx[i];
+                                struct connection cnx_i = *pcnx_i;
+                                for (i = 0; i < num_addr_listen; i++)
```

Comment:
The main_loop() function is already too long :)  -- I\d move this to a:
close_fds(int*, int)   => close_fds(listen_sockets, num_addr_listen);
close_cnx(struct connection*, int) => close_cnx(cnx, num_cnx);

or something like that.